department-of-veterans-affairs/caseflow
# Ki/APPEALS-49727 - Task Actions: Other motion: Mark Task Complete

Resolves [APPEALS-49727 - Task Actions: Other motion: Mark Task Complete](https://jira.devops.va.gov/browse/APPEALS-49727)

# Description
Created CorrespondenceCompleteTaskModal that populates on Details page.  This pops up after "Mark task complete" has been selected on an Other Motion task, allowing the task to be Completed.

## Acceptance Criteria

## Testing Plan
1. Navigate to a correspondence case review package page that does not have an intake already in progress(i.e. http://localhost:3000/queue/correspondence/053d5e15-716d-4bd7-8d7f-f6be737b52e3/review_package).
    1. NOTE: Save this URL(minus the review_package), you'll need it to navigate to the Correspondence Details after completing the intake.
2. Assign Correspondence Type.  Save changes.
3. Intake Correspondence with "Create Record" at bottom.
4. On Step 2, make sure you assign an "Other Motion" task to the correspondence.
5. Complete intake process.
6. Navigate to details page for that correspondence(i.e. http://localhost:3000/queue/correspondence/053d5e15-716d-4bd7-8d7f-f6be737b52e3/).
7. At the bottom of the first page, under the "Tasks not related to an appeal", find the Other Motion task you created.
8. Select "Mark task complete" under the dropdown action menu.
9. Click Cancel Link to test cancel.
10. Repeat dropdown.  Click X at top to test cancel.
11. Repeat dropdown.  Fill in the box on the Complete Modal and click "Mark Complete".
12. Task is removed from the "Tasks not related to an appeal" section, and a banner is displayed saying "Other motion task has been marked complete. If you have made a mistake, please email Litigation Support for any changes."

# Frontend
## User Facing Changes

### Front-End
![Screenshot 2024-08-28 at 10 48 13 AM](https://github.com/user-attachments/assets/331bcca9-4f41-4758-8174-7149be16f697)
![Screenshot 2024-08-28 at 10 48 22 AM](https://github.com/user-attachments/assets/f36dda3a-624a-4a2f-a071-ac4a48bd3272)
![Screenshot 2024-08-23 at 1 46 14 PM](https://github.com/user-attachments/assets/69d82976-b9b9-42d1-b742-2ef7b622a4ed)

### Dev/Supporting
![Screenshot 2024-08-21 at 4 49 35 PM](https://github.com/user-attachments/assets/98bc2206-a823-4224-a749-69039a4a01b5)
![Screenshot 2024-08-21 at 4 48 41 PM](https://github.com/user-attachments/assets/d9f648c5-c350-4c91-947a-0e180fe6ae6a)
![Screenshot 2024-08-23 at 1 46 46 PM](https://github.com/user-attachments/assets/afdd8a21-7340-4392-81a4-4cbd09c3b02c)

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:
- [X] Jest (Updated Case Details jest issues)
